### PR TITLE
Added Smart Charging Profile and SetChargingProfile.req / conf

### PR DESCRIPTION
**Pull request by emilm**

### CODE REVIEW

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/SetChargingProfileConfirmation.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public ChargingProfileStatus getStatus() {
+        return status;
+    }
+
+    /**
+     * This indicates the success or failure of the change of the charging profile.
+     * @return the {@link ChargingProfileStatus}.
+     */
+    @Deprecated
+    public ChargingProfileStatus objStatus() {
```
Comment (TVolden) [inline ocpp-v1_6/src/main/java/eu/chargetime/ocpp/model/smartcharging/SetChargingProfileConfirmation.java]: No need for this one. It was an old workaround for a bad serializer.

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public Feature[] getFeatureList() {
+        return features.toArray(new Feature[0]);
+    }
+
+    @Override
+    public Confirmation handleRequest(UUID sessionIndex, Request request) {
+       return null;
+    }
+
+    public SetChargingProfileRequest createSetChargingProfileRequest(int connectorId, ChargingProfile profile) throws PropertyConstraintException {
```
Comment (TVolden) [inline ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java]: Would be nice to have some Java doc notations.

File: ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileConfirmationTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+public class SetChargingProfileConfirmationTest {
+    SetChargingProfileConfirmation confirmation;
+
+    @Before
+    public void setUp() throws Exception {
+        confirmation = new SetChargingProfileConfirmation();
+    }
+
+    @Test
+    public void setIdTagInfo_aChargingProfileStatus_statusIsSet() {
```
Comment (TVolden) [inline ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileConfirmationTest.java]: I probably did this myself as well, but there is no need to test POJO methods if they don't have any validation. No harm though.

File: ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileConfirmationTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Test
+    public void validate_returnFalse() {
+        // When
+        boolean isValid = confirmation.validate();
+
+        // Then
+        assertThat(isValid, is(false));
+    }
+
```
Comment (TVolden) [inline ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileConfirmationTest.java]: Would be nice to have a positive test of the validate method.

File: ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileRequestTest.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import eu.chargetime.ocpp.PropertyConstraintException;
+import eu.chargetime.ocpp.model.smartcharging.SetChargingProfileRequest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class SetChargingProfileRequestTest {
```
Comment (TVolden) [inline ocpp-v1_6/src/test/java/eu/chargetime/ocpp/model/test/SetChargingProfileRequestTest.java]: Wouldn't really be complete before the validate method is tested.

File: ocpp-v1_6/src/test/java/eu/chargetime/ocpp/feature/profile/test/ClientSmartChargingProfileTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+        // When
+        smartCharging.handleRequest(SESSION_NULL, request);
+
+        // Then
+        verify(handler, times(1)).handleSetChargingProfileRequest(request);
+    }
+
+    @Test
+    public void handleRequest_SetChargingProfileRequest_returnsSetChargingProfileRequest() {
```
Comment (emilm) [inline ocpp-v1_6/src/test/java/eu/chargetime/ocpp/feature/profile/test/ClientSmartChargingProfileTest.java]: Fixed

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java (hunk 2 above)
Comment (emilm) [inline ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java]: Fixed

**Pull request merged**